### PR TITLE
feat: add user search screen

### LINKED ISSUE #42: Implement search screen and functionality
### **Objective:**

The goal is to create a search screen  component and it's functionality 

### **Description:**

- Create a component called `Search.tsx` and add it in `/components/ folder
- The must have ```CustomInput``` component that allows users to search by using username or name 
- When the username or name is found those friends must appeared on the screen they must be rendered with:
    - User profile avatar
    - Usename
    - Name 
    - ```ADD``` in the left right 
- When the ``ADD``` button is clicked on their must be indication that a friend is added with custom check box icon  
- The search component must have ```share``` component that allows a user to invite their friends via other social medias 
- Make the implementation match the updated Figma design
- This component must open when a user clicks on the ```FIND FREINDS``` on splash screen and also when a user clicks on the ```FIND```inside the ```Friends``` component 

### **Acceptance Criteria:**
- The component called `Search.tsx` is created and located in the above-specified folder location 
- The component matches the updated Figma design
- The component is Rendered in the above-specified folder location 

### **Design:**
Image
Image

### PR DESCRIPTION
**Pull request by igorntwari**
closes #42

### CODE REVIEW

File: components/Search.tsx [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+  const handleClose = () => {
+    navigation.goBack()
+  }
+
+  return (
+    <View style={styles.container}>
+      <View style={styles.header}>
+        <View style={styles.spacer} />
+        <Text style={styles.title}>Find Friends</Text>
```
Comment (Odile078) [inline components/Search.tsx:48]: let's use the customText here

File: components/Search.tsx [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+            ))
+          ) : (
+            <View style={styles.share}>
+              <ShareCard />
+            </View>
+          )
+        ) : (
+          <View style={styles.share}>
+            <ShareCard />
+          </View>
```
Comment (Odile078) [inline components/Search.tsx:93]: this card will be visible whether the user is searching or not, so it's better to just put it outside of these conditions

File: components/Search.tsx [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+import { availableFriends } from "@/data/users"
+import { useNavigation } from "@react-navigation/native"
+import UserAvatar from "@/components/ui/UserAvatar"
+
+const FindFriends = () => {
+  const navigation = useNavigation()
+  const [search, setSearch] = useState("")
+  const [addedFriends, setAddedFriends] = useState<{ [key: string]: boolean }>(
+    {},
+  )
```
Comment (Odile078) [inline components/Search.tsx:18]: isn't the `addedFriends`supposed to be an array of users?
```suggestion
  const [addedFriends, setAddedFriends] = useState<User[]>([])  )
```

File: components/Search.tsx [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+    </View>
+  )
+}
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    paddingHorizontal: 20,
+    paddingVertical: 70,
+    backgroundColor: "#fff",
```
Comment (Odile078) [inline components/Search.tsx:105]: let's use theme colors

File: components/Search.tsx [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+    paddingBottom: 22,
+  },
+  spacer: {
+    width: 24,
+  },
+  title: {
+    flex: 1,
+    textAlign: "center",
+    fontSize: 18,
+    fontWeight: "bold",
```
Comment (Odile078) [inline components/Search.tsx:120]: let's use fontSize and weight from the theme file

File: components/Search.tsx
Comment (Odile078) [inline components/Search.tsx:1]: can we have this in screens, since it's a screen?

File: screens/Search/Search.tsx
Comment (Odile078) [inline screens/Search/Search.tsx:1]: let's name this component index.tsx

File: components/ui/CheckBox.tsx [hunk 6]
```diff
@@ -24,10 +26,23 @@ export default function CheckBox({
...
+        style,
+      ]}
       onPress={handleCheckedChange}
       {...rest}>
-      {isChecked ? <CheckIcon /> : <PlusIcon />}
+      {isChecked ? (
+        <CheckIcon
+          color={unshaded ? theme.colors.black : theme.colors.white}
+          height={unshaded ? 50 : 32}
+          width={unshaded ? 50 : 32}
```
Comment (Odile078) [inline components/ui/CheckBox.tsx:41]: why is the height and width `50`, on the Figma it is `24*24`

File: screens/Search/Search.tsx [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+
+  const handleClose = () => {
+    navigation.goBack()
+  }
+
+  return (
+    <View style={styles.container}>
+      <View style={styles.header}>
+        <View style={styles.spacer} />
+        <CustomText fontFamily="suisse" size="lg" fontWeight="semibold">
```
Comment (Odile078) [inline screens/Search/Search.tsx:53]: the suisse fontFamily is the default font we don't need to apply it

File: components/ui/CheckBox.tsx (hunk 6 above)
Comment (igorntwari) [reply components/ui/CheckBox.tsx]: i thought 24 was small but let's keep it

**Review by Odile078 — APPROVED** [review]

File: screens/Search/index.tsx [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+                </View>
+              </View>
+              {addedFriends.some(
+                (addedFriend) => addedFriend.id === friend.id,
+              ) ? (
+                <CheckBox
+                  isChecked={true}
+                  unshaded
+                  onPress={() => handleAddFriend(friend)}
+                />
```
Comment (serapieTuyishime) [inline screens/Search/index.tsx:91]: use `<CheckIcon />`.

File: components/ui/CheckBox.tsx
Comment (serapieTuyishime) [inline components/ui/CheckBox.tsx:1]: Then these changes won't be needed

**Review by serapieTuyishime — CHANGES_REQUESTED** [review]

**Review by serapieTuyishime — APPROVED** [review]

**Review by AnasAlamin — APPROVED** [review]

**Pull request merged**